app/services/csv/lettings_log_csv_service.rb

```diff
@@ -272,34 +272,43 @@ def yes_or_blank_label(value)
-      "housingneeds_type" => %w[housingneeds_type housingneeds_a housingneeds_b housingneeds_c housingneeds_f housingneeds_g housingneeds_h],
-      "net_income_known" => %w[net_income_known incref],
-      "irproduct_other" => %w[irproduct irproduct_other lar],
-      "la" => %w[is_la_inferred la_label la],
-      "prevloc" => %w[is_previous_la_inferred prevloc_label prevloc],
-      "needstype" => %w[needstype lettype],
-      "prevten" => %w[prevten new_old],
-      "voiddate" => %w[voiddate vacdays],
-      "rsnvac" => %w[rsnvac newprop],
-      "household_charge" => %w[household_charge nocharge],
-      "brent" => %w[brent wrent rent_value_check],
-      "scharge" => %w[scharge wscharge],
-      "pscharge" => %w[pscharge wpschrge],
-      "supcharg" => %w[supcharg wsupchrg],
-      "tcharge" => %w[tcharge wtcharge],
-      "chcharge" => %w[chcharge wchchrg],
-      "tshortfall" => %w[tshortfall wtshortfall],
-      "letting_allocation_unknown" => %w[letting_allocation_none],
-    }.freeze
+    ORDERED_ADDRESS_FIELDS = %w[uprn address_line1 address_line2 town_or_city county postcode_full is_la_inferred la_label la uprn_known uprn_selection address_search_value_check address_line1_input postcode_full_input address_line1_as_entered address_line2_as_entered town_or_city_as_entered county_as_entered postcode_full_as_entered la_as_entered].freeze
+
+    def attribute_mappings
+      mappings = {
+        "owning_organisation_id" => %w[owning_organisation_name],
+        "managing_organisation_id" => %w[managing_organisation_name],
+        "assigned_to_id" => [],
+        "scheme_id" => [],
+        "location_id" => [],
+        "rent_type" => %w[renttype renttype_detail],
+        "hb" => %w[hb has_benefits],
+        "age1" => %w[refused hhtype totchild totelder totadult age1],
+        "housingneeds_type" => %w[housingneeds_type housingneeds_a housingneeds_b housingneeds_c housingneeds_f housingneeds_g housingneeds_h],
+        "net_income_known" => %w[net_income_known incref],
+        "irproduct_other" => %w[irproduct irproduct_other lar],
+        "la" => %w[is_la_inferred la_label la],
+        "prevloc" => %w[is_previous_la_inferred prevloc_label prevloc],
+        "needstype" => %w[needstype lettype],
+        "prevten" => %w[prevten new_old],
+        "voiddate" => %w[voiddate vacdays],
+        "rsnvac" => %w[rsnvac newprop],
+        "household_charge" => %w[household_charge nocharge],
+        "brent" => %w[brent wrent rent_value_check],
+        "scharge" => %w[scharge wscharge],
+        "pscharge" => %w[pscharge wpschrge],
+        "supcharg" => %w[supcharg wsupchrg],
+        "tcharge" => %w[tcharge wtcharge],
+        "chcharge" => %w[chcharge wchchrg],
+        "tshortfall" => %w[tshortfall wtshortfall],
+        "letting_allocation_unknown" => %w[letting_allocation_none],
+      }
+
+      if @user.support? && @year >= 2024
+        mappings["declaration"] = %w[declaration] + ORDERED_ADDRESS_FIELDS
```

Comment:
Similar to comment on sales - this looks a bit unintuitive to me, although I'm not sure if adjusting it would actually make things better overall.

Reply:
Is it a bit better now?

Reply:
Yeah, that works